### PR TITLE
dispose of controller status listener

### PR DESCRIPTION
**Pull request by jayjah**

### CODE REVIEW

**Review by adrinr — APPROVED** [review]

**Review by mdeandrea-mrmilu — APPROVED** [review]

File: lib/shimmer.dart [hunk 1]
```diff
@@ -125,29 +125,31 @@ class Shimmer extends StatefulWidget {
...
-          _controller.forward(from: 0.0);
-        }
-      });
+      ..addStatusListener(controllerStateListener);
     if (widget.enabled) {
       _controller.forward();
     }
   }
 
+  void controllerStateListener(AnimationStatus status) {
```
Comment (hnvn) [inline lib/shimmer.dart:141]: This method should be private and the method name should start with underscore (as convention)

**Review by hnvn — COMMENTED** [review]
There's a small problem about coding convention

File: lib/shimmer.dart (hunk 1 above)
Comment (jayjah) [reply lib/shimmer.dart]: is Done! Thx for hint

**Comment by hasanm08** [comment]
@adrinr 
@hnvn 
@mdeandrea-mrmilu 
@vasilich6107 
merge please

**Comment by hnvn** [comment]
We actually don't need to call `removeStatusListener` because `dispose` method of `AnimationController` already took care of cleaning all listeners for us. You can see it in codes of `dispose` function: https://api.flutter.dev/flutter/animation/AnimationController/dispose.html

**Comment by hasanm08** [comment]
> We actually don't need to call `removeStatusListener` because `dispose` method of `AnimationController` already took care of cleaning all listeners for us. You can see it in codes of `dispose` function: https://api.flutter.dev/flutter/animation/AnimationController/dispose.html

I checked both his branch and your latest version
in his version flutter html web renderer issue fixed
but in version 3.0.0 still exists
try

```
dependency_overrides:
  shimmer:
    git:
      url: https://github.com/jayjah/flutter_shimmer.git
      ref: master
```

**Comment by hasanm08** [comment]
#75

**Pull request closed without merging**